### PR TITLE
Active Admin Dashboard is semi-finished.

### PR DESCRIPTION
**Pull request by starkshaw**
The known problem is:
- Current dashboard will conflict with admin dashboard, reason unknown. To fix that problem, I replaced `dashboard_path` to `root_path` on `def after_sign_in_path_for(resource)`.
- After logged into the Active Admin, it diverts to the `root` page rather than `admin/`, and somehow the `root` page requires log-in. But if the super user manually filled `admin/` in URL it will leads to the Admin Dashboard.

### CODE REVIEW

File: app/admin/dashboard.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  content title: proc{ I18n.t("active_admin.dashboard") } do
+    div class: "blank_slate_container", id: "dashboard_default_message" do
+      span class: "blank_slate" do
+        span I18n.t("active_admin.dashboard_welcome.welcome")
+        small I18n.t("active_admin.dashboard_welcome.call_to_action")
+      end
+    end
+
+    # Here is an example of a simple dashboard with columns and panels.
```
Comment (danielpaul) [inline app/admin/dashboard.rb]: Could we get some dashboard widgets here? Like latest users, new shops, products and sales? Have a look at LifeBites dashboard.

File: app/admin/product.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+ActiveAdmin.register Product do
+
+# See permitted parameters documentation:
+# https://github.com/activeadmin/activeadmin/blob/master/docs/2-resource-customization.md#setting-up-strong-parameters
+#
+# permit_params :list, :of, :attributes, :on, :model
```
Comment (danielpaul) [inline app/admin/product.rb]: Let's get some real forms and things in here. Same for all the other models too. Ability to edit, delete and view and show information that are the most useful to the marketplace owner. Like for example when the marketplace owner is looking at a single shop page in the admin, show all the products in the shop, and in the products, show all the sales of those products, etc.

File: config/routes.rb [hunk 3]
```diff
@@ -1,5 +1,7 @@
 Rails.application.routes.draw do
 
+  devise_for :admin_users, ActiveAdmin::Devise.config
```
Comment (danielpaul) [inline config/routes.rb]: Could we add this further down the routes file? Just to keep everything organised. You might do a clean up of the routes file when are you on it. Like for example the rate route should be further down.

File: db/migrate/20160307110825_devise_create_admin_users.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      t.inet     :current_sign_in_ip
+      t.inet     :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
+
+      ## Lockable
```
Comment (danielpaul) [inline db/migrate/20160307110825_devise_create_admin_users.rb:28]: Let's enable lockable for the admin user. That would involve commenting this out, adding it to the admin user model and resetting db.

File: db/seeds.rb [hunk 5]
```diff
@@ -5,3 +5,4 @@
 #
 #   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
 #   Mayor.create(name: 'Emanuel', city: cities.first)
+AdminUser.create!(email: 'admin@example.com', password: 'password', password_confirmation: 'password')
```
Comment (danielpaul) [inline db/seeds.rb]: By default set this to be more unpredictable.

**Comment by danielpaul** [comment]
@LevinGeorge Could you also look at the `.hound.yml` config again and remove the lines too long messages please?

**Comment by starkshaw** [comment]
@danielpaul In terms of that `.hound.yml` I configured but didn't work... Its documentation doesn't cover that part.

**Comment by LevinGeorge** [comment]
@danielpaul I'm going to tweet them and see. @starkshaw googled it all weekend and couldn't find anything

**Comment by danielpaul** [comment]
@LevinGeorge Could we try extending the max line count and see if that would make any difference?

**Comment by LevinGeorge** [comment]
@danielpaul yeah!

**Comment by starkshaw** [comment]
@danielpaul https://github.com/SprintFire/marketplace/blob/master/.hound.yml This is what I pushed on Saturday, found online, didn't work.

File: app/admin/product.rb (hunk 2 above)
Comment (starkshaw) [inline app/admin/product.rb]: I don't understand this part, please elaborate?

**Comment by danielpaul** [comment]
@starkshaw Let's add `.ruby-style.yml` file which might fix it. 

https://houndci.com/configuration
https://github.com/houndci/hound/blob/master/config/style_guides/ruby.yml
https://twitter.com/nwkr/status/663556461855834112

**Comment by starkshaw** [comment]
@danielpaul Please review.

File: app/admin/product.rb (hunk 2 above)
Comment (danielpaul) [inline app/admin/product.rb]: Need the products to be editable. Need to configure this file. Show sales of each products too and the shop they belong to.

File: app/admin/purchase.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+ActiveAdmin.register Purchase do
```
Comment (danielpaul) [inline app/admin/purchase.rb:1]: Show product details on the purchases.

File: app/admin/shop.rb [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        # Wait until the payment integration is compeleted.
+      end
+    end
+    active_admin_comments
+  end
+
+  form do |f|
+    f.inputs "User Details" do
+      f.input :email
+      f.input :first_name
```
Comment (danielpaul) [inline app/admin/shop.rb]: Errmm... this should be related to this model?

File: app/admin/user.rb [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  form do |f|
+    f.inputs "User Details" do
+      f.input :email
+      f.input :first_name
+      f.input :last_name
+    end
+    f.actions
+  end
+
+# See permitted parameters documentation:
```
Comment (danielpaul) [inline app/admin/user.rb]: You can delete these comments when you don't need them.

File: app/admin/user.rb [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+      row :created_at
+      row :updated_at
+      row :current_sign_in_at
+      row :last_sign_in_at
+      row :current_sign_in_ip
+      row :last_sign_in_ip
+      row :sign_in_count
+    end
+
+    panel "Purchase History" do
```
Comment (danielpaul) [inline app/admin/user.rb:48]: Other things to show would be the shops they own.

File: db/migrate/20160307110825_devise_create_admin_users.rb [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+      t.datetime :locked_at
+
+
+      t.timestamps null: false
+    end
+
+    add_index :admin_users, :email,                unique: true
+    add_index :admin_users, :reset_password_token, unique: true
+    # add_index :admin_users, :confirmation_token,   unique: true
+    # add_index :admin_users, :unlock_token,         unique: true
```
Comment (danielpaul) [inline db/migrate/20160307110825_devise_create_admin_users.rb]: Need to comment this out for lockable

File: app/admin/shop.rb (hunk 7 above)
Comment (starkshaw) [inline app/admin/shop.rb]: Just fixed, forgot to change...

File: app/admin/purchase.rb (hunk 6 above)
Comment (starkshaw) [inline app/admin/purchase.rb:1]: Finished.

File: app/admin/user.rb (hunk 9 above)
Comment (starkshaw) [inline app/admin/user.rb:48]: Finished.

File: app/admin/user.rb (hunk 8 above)
Comment (starkshaw) [inline app/admin/user.rb]: For future reference... :p Are they not being minimized?

File: db/migrate/20160307110825_devise_create_admin_users.rb (hunk 10 above)
Comment (starkshaw) [inline db/migrate/20160307110825_devise_create_admin_users.rb]: I thought I already did?

Comment (starkshaw) [inline db/migrate/20160307110825_devise_create_admin_users.rb]: Oh I got you.

File: app/admin/dashboard.rb [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+            column(:buyer_email)   {|purchase| link_to purchase.user.email, admin_user_path(purchase.user) }
+            column(:product) {|purchase| link_to purchase.product.name, admin_product_path(purchase.product) }
+            column("From Shop") {|purchase| link_to purchase.product.shop.name, admin_shop_path(purchase.product.shop) }
+            column("Purchased at")    {|purchase| time_ago_in_words(purchase.created_at) + " ago" }
+          end
+        end
+      end
+    end
+
+    # Here is an example of a simple dashboard with columns and panels.
```
Comment (danielpaul) [inline app/admin/dashboard.rb]: Remove these comments?

File: app/admin/product.rb (hunk 2 above)
Comment (danielpaul) [inline app/admin/product.rb]: This still needs to be done.

File: app/admin/purchase.rb (hunk 6 above)
Comment (danielpaul) [inline app/admin/purchase.rb:1]: Looks good :) :+1:

File: app/admin/purchase.rb [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+        column("Product Name") {|product| product.name}
+        column("Listed Date", :sortable => :created_at){|product| pretty_format(product.created_at) }
+        column("Update Date", :sortable => :updated_at){|product| pretty_format(product.updated_at) }
+        column("Current Price") {|product| number_to_currency product.price}
+        column("From Shop") {|product| link_to product.shop.name, admin_shop_path(product.shop)}
+      end
+    end
+  end
+
+# See permitted parameters documentation:
```
Comment (danielpaul) [inline app/admin/purchase.rb]: Remove these comments too pls.

File: app/admin/user.rb (hunk 8 above)
Comment (danielpaul) [inline app/admin/user.rb]: What do you mean? Server side code doesn't get minified. Comments don't matter but we need to have a clean codebase to maintain without a lot of junk comments.

File: db/migrate/20160307110825_devise_create_admin_users.rb (hunk 4 above)
Comment (danielpaul) [inline db/migrate/20160307110825_devise_create_admin_users.rb:28]: You still need to enable lockable.

**Comment by danielpaul** [comment]
@starkshaw There are still some feedback on this that aren't fixed yet. https://github.com/SprintFire/marketplace/pull/56 Check the comments on the code that are still remaining.

**Comment by starkshaw** [comment]
@danielpaul Yeah I'm fixing it. The Lockable is uncommented.

**Comment by danielpaul** [comment]
@starkshaw You need to add `:lockable` to `app/models/admin_user.rb`

**Comment by starkshaw** [comment]
Finished, please review.

File: app/models/admin_user.rb [hunk 13]
```diff
@@ -0,0 +1,6 @@
+class AdminUser < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable,
+         :recoverable, :rememberable, :trackable, :validatable
```
Comment (danielpaul) [inline app/models/admin_user.rb]: Need to add `: lockable` to here.

File: db/seeds.rb (hunk 5 above)
Comment (danielpaul) [inline db/seeds.rb]: Also need to change this default to something more unpredictable. Just to be secure by default :P

Comment (starkshaw) [inline db/seeds.rb]: Just found this by unfold all comments...

**Comment by starkshaw** [comment]
Finished, please review.

**Pull request merged**